yellowbox_heksher/heksher_service.py

```diff
@@ -14,24 +15,33 @@
 
 class HeksherService(SingleEndpointService, RunMixin):
     def __init__(self, docker_client: DockerClient, postgres_image: str = 'postgres:latest',
-                 heksher_image: str = 'biocatchltd/heksher:0.3.0', port: int = 0, *,
+                 heksher_image: str = 'biocatchltd/heksher:0.4.1', port: int = 0, *,
                  heksher_startup_context_features: str, **kwargs):
         creator = SafeContainerCreator(docker_client)
 
         self.postgres_service = PostgreSQLService(docker_client, image=postgres_image, default_db='heksher')
 
+        self.heksher_db_migrator = creator.create_and_pull(
+            heksher_image, "alembic upgrade head", publish_all_ports=True, ports={80: port}, detach=True, environment={
+                'HEKSHER_DB_CONNECTION_STRING': self.postgres_service.container_connection_string("postgres"),
+                'HEKSHER_STARTUP_CONTEXT_FEATURES': heksher_startup_context_features,
+            }
+        )
+
         self.heksher = creator.create_and_pull(
             heksher_image, publish_all_ports=True, ports={80: port}, detach=True, environment={
                 'HEKSHER_DB_CONNECTION_STRING': self.postgres_service.container_connection_string("postgres"),
                 'HEKSHER_STARTUP_CONTEXT_FEATURES': heksher_startup_context_features,
             }
         )
+
         self.http_client: Optional[httpx.AsyncClient] = None
 
         self.network = anonymous_network(docker_client)
         self.postgres_service.connect(self.network, aliases=['postgres'])
         self.network.connect(self.heksher, aliases=['heksher'])
-        super().__init__((self.heksher, ), **kwargs)
+        self.network.connect(self.heksher_db_migrator, aliases=['heksher-db-migrator'])
+        super().__init__((self.heksher_db_migrator, self.heksher), **kwargs)
 
     def start(self, retry_spec: Optional[RetrySpec] = None):
```

Comment:
you should wait until the migration is done before you run the main heksher service.